Flaky-test: ManagedLedgerBkTest.verifyConcurrentUsage

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Example failure

https://github.com/codelipenghui/incubator-pulsar/actions/runs/3162222563/jobs/5148685244

### Exception stacktrace

```
  Error:  Tests run: 14, Failures: 1, Errors: 0, Skipped: 2, Time elapsed: 12.45 s <<< FAILURE! - in org.apache.bookkeeper.mledger.impl.ManagedLedgerBkTest
  Error:  verifyConcurrentUsage(org.apache.bookkeeper.mledger.impl.ManagedLedgerBkTest)  Time elapsed: 1.058 s  <<< FAILURE!
  java.lang.AssertionError: expected [0.0] but found [128.0]
  	at org.testng.Assert.fail(Assert.java:99)
  	at org.testng.Assert.failNotEquals(Assert.java:1037)
  	at org.testng.Assert.assertEquals(Assert.java:701)
  	at org.testng.Assert.assertEquals(Assert.java:712)
  	at org.apache.bookkeeper.mledger.impl.ManagedLedgerBkTest.verifyConcurrentUsage(ManagedLedgerBkTest.java:224)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
  	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at java.base/java.lang.Thread.run(Thread.java:833)
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/RangeCache.java
+++ b/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/RangeCache.java
@@ -196,13 +196,11 @@ public class RangeCache<Key extends Comparable<Key>, Value extends ReferenceCoun
            if (entry == null || timestampExtractor.getTimestamp(entry.getValue()) > maxTimestamp) {
                break;
            }
-
-           entry = entries.pollFirstEntry();
-           if (entry == null) {
+           Value value = entries.remove(entry.getKey());
+           if (value == null) {
                break;
            }
 
-           Value value = entry.getValue();
            removedSize += weighter.getSize(value);
            removedCount++;
            value.release();

--- a/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/RangeCache.java
+++ b/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/RangeCache.java
@@ -196,8 +196,9 @@ public class RangeCache<Key extends Comparable<Key>, Value extends ReferenceCoun
            if (entry == null || timestampExtractor.getTimestamp(entry.getValue()) > maxTimestamp) {
                break;
            }
-           Value value = entries.remove(entry.getKey());
-           if (value == null) {
+           Value value = entry.getValue();
+           boolean removeHits = entries.remove(entry.getKey(), value);
+           if (!removeHits) {
                break;
            }
 

